File: A_Twins.py
def main():
    _ = int(input())
    coins: list[int] = sorted(map(int, input().split()), reverse=True)
    sum_of_all_coins: int = sum(coins)
    partial_sum: int = 0

    for index, coin in enumerate(coins):
        partial_sum += coin
        if partial_sum > sum_of_all_coins // 2:
            print(index + 1)
            break

if __name__ == "__main__":
    main()

File: A_Twins.rs
use std::io::BufRead;

fn main() {
    let mut stdin = std::io::stdin().lock();
    let mut read_line = || {
        let mut line = String::new();
        stdin.read_line(&mut line).unwrap();
        line
    };

    _ = read_line();
    let mut coins: Vec<u8> = read_line()
        .split_ascii_whitespace()
        .map(|x| x.parse().unwrap())
        .collect();
    coins.sort_by(|a, b| b.cmp(a));
    let sum_of_all_coins: u16 = coins.iter().fold(0, |acc, x| acc + (*x as u16));
    let mut partial_sum: u16 = 0;

    for (index, coin) in coins.iter().enumerate() {
        partial_sum += (*coin as u16);
        if partial_sum > sum_of_all_coins / 2 {
            println!("{}", index + 1);
            break;
        }
    }
}